### DISCUSSION TITLE
Is there any way to change stream data length and ID? (HTTP2 + GRPC)

### DISCUSSION THREAD

**Discussion by daniel-rome** [mitmproxy/mitmproxy#5657] [Q&A]
I was able to modify the GRPC data by using the "flow.response.stream" function, but there are bytes before such data that mark the number of data bytes (length) and stream ID that I was not able to modify. Is there any way to do it?

For example, if stream data is 4 bytes and I modify them to 3 bytes, how could I set the new GRPC length on the stream response?

Thanks.

**Comment by mhils** [comment]
Modifying streamed messages never really works well unless you only want to flip a character or so. If possible, don't use streaming and replace on `flow.response.content`.

**Reply by daniel-rome** [reply]
Modifying stream data via (flow.response.stream) worked me quite well in my last tests. The problem is when I want to change (through the data) the size of such response. "flow.response.content" returned me "None", is there any way to modify the full response in raw?
